adapters/faraday/lib/opentelemetry/adapters/faraday/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Faraday
+      module Middlewares
+        class TracerMiddleware < ::Faraday::Middleware
+          def call(env)
+            tracer.in_span(env.url.to_s,
+                          attributes: { 'component' => 'http',
+                                        'http.method' => env.method,
```

Comment:
dd-trace-rb has the target host and port as attributes: https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/contrib/faraday/middleware.rb#L45-L46. 

I think we should strive the keep the same attributes and see if there are analogs in the semantic conventions. If they are absent from the semantic conventions, perhaps we should consider opening a spec issue or PR to add them.

Reply:
[Semantic conventions for HTTP client](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md#http-client) state:

> One of the following sets of attributes is required (in order of usual preference unless for a particular web client/framework it is known that some other set is preferable for some reason; all strings must be non-empty):
>
> * http.url
> * http.scheme, http.host, http.target
> * http.scheme, peer.hostname, peer.port, http.target
> * http.scheme, peer.ip, peer.port, http.target

So I think just recording `http.url` is preferred.

Reply:
WFM 👍